### ISSUE TITLE
Crash when uploading a large file

### ISSUE THREAD

**Issue by tianyuanhao** [tsl0922/ttyd#278]
**Describe the bug**

> uv_try_write: EAGAIN

`ttyd` crashes when using `rz` to upload a large (> 16 K) file.
**1.6.0** has this problem but **1.5.2** does not.

**Environment:**

- OS: Ubuntu 16.04
- Browser: Chrome 77

**Additional context**

Hope to solve [this problem](https://github.com/tsl0922/ttyd/issues/99#issuecomment-504816527) together.

**Comment by Zibri** [comment]
also with small files like a jpg is locks in infinite loop and need a refresh/relogin
same goes with sz

**Comment by tsl0922** [comment]
libwebsockets version? it's printed on ttyd start.

> [2020/03/04 11:50:02:7620] N: ttyd 1.6.0 (libwebsockets 3.2.2)

**Comment by tianyuanhao** [comment]
> libwebsockets version? it's printed on ttyd start.

Both **3.2.0** and **3.2.2** have been tested. Same error.

**Comment by tianyuanhao** [comment]
@tsl0922 This (yodaos-project/ShadowNode#223) might give you some ideas.

**Comment by tsl0922** [comment]
@tianyuanhao I need the stack trace. If you can reproduce it, use gdb to get it.

**Comment by tianyuanhao** [comment]
```
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
[2020/03/05 01:32:15:0262] N: ttyd 1.6.0 (libwebsockets 3.2.0)
[2020/03/05 01:32:15:0262] N: tty configuration:
[2020/03/05 01:32:15:0262] N:   start command: bash
[2020/03/05 01:32:15:0263] N:   close signal: SIGHUP (1)
[2020/03/05 01:32:15:0263] N:   terminal type: xterm-256color
[2020/03/05 01:32:15:0315] N:  Using foreign event loop...
[2020/03/05 01:32:15:0316] N:  Listening on port: 7681
[2020/03/05 01:32:18:8242] N: HTTP / - 172.17.0.1
[2020/03/05 01:32:19:3579] N: HTTP /token - 172.17.0.1
[2020/03/05 01:32:19:3645] N: WS   /ws - 172.17.0.1, clients: 1
[Detaching after fork from child process 4167]
[2020/03/05 01:32:19:5196] N: started process, pid: 4167
[2020/03/05 01:32:25:1789] E: uv_try_write: EAGAIN
[2020/03/05 01:32:25:1790] N: WS closed from 172.17.0.1, clients: 0
[2020/03/05 01:32:25:1790] N: killing process 4167 with signal: 1 (SIGHUP)
[2020/03/05 01:32:25:1798] N: process killed with signal 1 (SIGHUP), pid: 4167
[2020/03/05 01:32:25:2856] N: HTTP /token - 172.17.0.1
[2020/03/05 01:32:25:2948] N: WS   /ws - 172.17.0.1, clients: 1
[Detaching after fork from child process 4171]
[2020/03/05 01:32:25:3648] N: started process, pid: 4171
```

There is something wrong with my statement. `ttyd` itself did not crash, but the web page crashed.

**Comment by tianyuanhao** [comment]
@tsl0922 Here is the stack trace.

```
Reading symbols from ./ttyd...
(gdb) b src/protocol.c:376
Breakpoint 1 at 0x1265b: file /ttyd/src/protocol.c, line 376.
(gdb) r bash
Starting program: /ttyd/build/ttyd bash
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
[2020/03/05 01:51:01:5571] N: ttyd 1.6.0 (libwebsockets 3.2.0)
[2020/03/05 01:51:01:5571] N: tty configuration:
[2020/03/05 01:51:01:5571] N:   start command: bash
[2020/03/05 01:51:01:5571] N:   close signal: SIGHUP (1)
[2020/03/05 01:51:01:5571] N:   terminal type: xterm-256color
[2020/03/05 01:51:01:5624] N:  Using foreign event loop...
[2020/03/05 01:51:01:5625] N:  Listening on port: 7681
[2020/03/05 01:51:04:5679] N: HTTP / - 172.17.0.1
[2020/03/05 01:51:04:0993] N: HTTP /token - 172.17.0.1
[2020/03/05 01:51:04:1018] N: WS   /ws - 172.17.0.1, clients: 1
[Detaching after fork from child process 4384]
[2020/03/05 01:51:05:2927] N: started process, pid: 4384

Breakpoint 1, callback_tty (wsi=0x5555555cbd80, reason=LWS_CALLBACK_RECEIVE, user=0x5555555c7920, in=0x5555555c8d50, len=4094) at /ttyd/src/protocol.c:376
376	                        lwsl_err("uv_try_write: %s\n", uv_err_name(err));
(gdb) bt
#0  callback_tty (wsi=0x5555555cbd80, reason=LWS_CALLBACK_RECEIVE, user=0x5555555c7920, in=0x5555555c8d50, len=4094) at /ttyd/src/protocol.c:376
#1  0x000055555556f383 in user_callback_handle_rxflow ()
#2  0x0000555555579395 in lws_parse_ws ()
#3  0x0000555555573e7e in lws_read_h1 ()
#4  0x00005555555751f2 in rops_handle_POLLIN_ws ()
#5  0x000055555556e6d7 in lws_service_fd_tsi ()
#6  0x000055555558fd2b in lws_io_cb ()
#7  0x00007ffff7c1fab0 in uv.io_poll () from /lib/x86_64-linux-gnu/libuv.so.1
#8  0x00007ffff7c0f7ac in uv_run () from /lib/x86_64-linux-gnu/libuv.so.1
#9  0x000055555556e908 in lws_service ()
#10 0x0000555555563e1c in main (argc=2, argv=0x7fffffffe758) at /ttyd/src/server.c:514
```

**Comment by tianyuanhao** [comment]
Similar to this [issue](https://github.com/FGasper/zmodemjs/issues/5#issuecomment-504735706).

**Comment by tsl0922** [comment]
should be fixed by 7118186b670122e0fda94a2c3c3da4a015b2b21b.
